server.go

```diff
@@ -490,21 +490,22 @@ func (server *Server) createTLSConfig(entryPointName string, tlsOption *TLS, rou
 	return config, nil
 }
 
-func (server *Server) startServer(srv *manners.GracefulServer, globalConfiguration GlobalConfiguration) {
+func (server *Server) startServer(srv *http.Server, globalConfiguration GlobalConfiguration) {
 	log.Infof("Starting server on %s", srv.Addr)
 	if srv.TLSConfig != nil {
-		if err := srv.ListenAndServeTLSWithConfig(srv.TLSConfig); err != nil {
+		err := srv.ListenAndServeTLS("", "")
```

Comment:
You could declare the `err` variable at the top, assign it in each TLS/no-TLS case, and check for `nil` once only:

```go
var err error
if srv.TLSConfig != nil {
  err = srv.ListenAndServeTLS("", "")
} else {
  err = srv.ListenAndServe()
}

if err != nil {
  log.Fatal("Error creating server: ", err)
}
```

Saves us one nested level of `if`ing.